api/client/src/test/java/org/projectnessie/client/auth/oauth2/TestOAuth2Client.java

```diff
@@ -891,4 +899,10 @@ private static AtomicReference<Runnable> mockTokensRefreshSchedule(
             });
     return task;
   }
+
+  private static TokensResponseBase asResponse(Tokens tokens) throws Exception {
+    Field enclosingInstance = tokens.getClass().getDeclaredField("this$0");
```

Comment:
Admittedly a little brittle, but works.

Reply:
Hacky - but since it's a test, that works for me.